dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1479,180 +1465,189 @@ synchronized String evaluateInterval() {
         long deltaTime = Math.max(currentTime - lastTimerPop, interval);
         long deltaCompleted = completedWork - previousCompleted;
         double throughput = 1000.0 * deltaCompleted / deltaTime;
+        queueDepth = threadPool.getQueue().size();
+        activeThreads = threadPool.getActiveCount();
+        boolean queueEmpty = (queueDepth <= 0);
+        // Count the number of consecutive times we've seen an empty queue
+        if (!queueEmpty) {
+            consecutiveQueueEmptyCount = 0;
+        } else if (lastAction != LastAction.SHRINK) { // 9/5/2012
+            consecutiveQueueEmptyCount++;
+        }
+
+        ThroughputDistribution currentStats = getThroughputDistribution(poolSize, true);
+
+        double forecast = currentStats.getMovingAverage();
```

Comment:
does the forecast have value in the trace statement if core threads and max threads are equal, or should this be 0 like is done for the shrinkScore and growScore and move this logic back inside of the if statement only.

Reply:
Good point. Changed to initialize forecast =0 and moved the `currentStats` lines inside the if statement.

Reply:
looks like that throughput distribution is used in the interval data though so it possibly is needed to be done outside of the if statement.  So I think it may make sense to still put it into the if statement like it was but change possibly to have something like:

````
if (coreThreads == maxThreads) {
     getThroughputDistribution(poolSize, true);
} else {
```
We don't need to use the return value.